jbpm-system-test/jbpm-smartrouter-test/src/test/java/org/kie/samples/integration/SmartRouterSystemTest.java

```diff
@@ -0,0 +1,445 @@
+    }
+    
+    @BeforeEach
+    private void before() throws Exception {
+        logger.debug("*** Before each ***");
+        
+        consumerSmartRouter = new WaitingConsumer();
+        smartRouter.followOutput(consumerSmartRouter, STDERR);
+        
+        await().atMost(ONE_MINUTE).untilAsserted(() -> assertThat(readNumberOfContainers()).isEqualTo(2));
+
+        initConfigFile = readConfigFileFromSmartRouter();
+        logger.debug("----------------> initConfigFile:: "+initConfigFile);
+        
+        String logs = smartRouter.getLogs(STDERR);
+        reachedLimitCounter = StringUtils.countMatches(logs, REACHED_ATTEMPTS_LIMIT);
+    }
+    
+    @AfterEach
+    private void writeConfigFileToContainer() throws Exception {
+        logger.debug("*** After each: writeConfigFileToContainer ***");
+        
+        removeAllToxics(proxy3);
+        removeAllToxics(proxy1);
+        
+        String logs = smartRouter.getLogs(STDERR);
+        reachedLimitCounter = StringUtils.countMatches(logs, REACHED_ATTEMPTS_LIMIT);
+        
+        Path path = Files.write(
+                Paths.get("kie-server-router.json"),
+                initConfigFile.getBytes(StandardCharsets.UTF_8));
+        //Need to give read/write permissions for all users -0666-
+        final MountableFile mountableFile = MountableFile.forHostPath(path, 0666);
+        smartRouter.copyFileToContainer(mountableFile, "opt/jboss/");
+        
+        await().atMost(ONE_MINUTE).until(() -> assertServersInRoutingTable(URL_NODE1, URL_NODE3));
+    }
+    
+    
+    private int readNumberOfContainers() throws JSONException {
+        JSONObject containers = getContainersFromSmartRouter();
+        return containers.has("result") ? 
+               containers.getJSONObject("result").getJSONObject("kie-containers").getJSONArray("kie-container").length() : 0;
+    }
+
+
+    private void removeAllToxics(ToxiproxyContainer.ContainerProxy proxy) throws IOException {
+        for (Toxic t : proxy.toxics().getAll()) {
+            t.remove();
+        }
+    }
+    
+    
+    private static Stream<Arguments> provideToxics() throws IOException {
+        ToxicSupplier<Toxic, IOException> limitData1 = () -> proxy1.toxics().limitData("limitData", DOWNSTREAM, 5);
+        ToxicSupplier<Toxic, IOException> resetPeer1 = () -> proxy1.toxics().resetPeer("resetPeer", DOWNSTREAM, 0);
+        ToxicSupplier<Toxic, IOException> timeout1 = () -> proxy1.toxics().timeout("timeout", DOWNSTREAM, getRandomTimeout(0,3000));
+        
+        ToxicSupplier<Toxic, IOException> limitData3 = () -> proxy3.toxics().limitData("limitData", DOWNSTREAM, 5);
+        ToxicSupplier<Toxic, IOException> resetPeer3 = () -> proxy3.toxics().resetPeer("resetPeer", DOWNSTREAM, 0);
+        ToxicSupplier<Toxic, IOException> timeout3 = () -> proxy3.toxics().timeout("timeout", DOWNSTREAM, getRandomTimeout(2000,5000));
+        
+        return Stream.of(Arguments.of(limitData1, null),
```

Comment:
Maybe for clarity purposes, I would add some name to each test, so it's easier to find which one failed or succeeded by using some `Named.of("Test1")` provided by junit5

Reply:
Cannot use it yet as this Named interface is since 5.8.
Anyway, it's also displayed in debug level when printing "Exercising with toxics:{}"